### PR TITLE
Adding support for MariaDB.

### PR DESCRIPTION
**Pull request by renemadsen**
Adding support for MariaDBs MARIADB_ROOT_PASSWORD.

### CODE REVIEW

File: src/Testcontainers/Configurations/Modules/Databases/MariaDbTestcontainerConfiguration.cs [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+  /// <inheritdoc cref="TestcontainerDatabaseConfiguration" />
+  [PublicAPI]
+  public class MariaDbTestcontainerConfiguration : TestcontainerDatabaseConfiguration
+  {
+    private const string MariaDbImage = "mariadb:10.4.10";
+
+    private const int MariaDbPort = 3306;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="MySqlTestcontainerConfiguration" /> class.
```
Comment (HofmeisterAn) [inline src/Testcontainers/Configurations/Modules/Databases/MariaDbTestcontainerConfiguration.cs]: `<see cref="MariaDbTestcontainerConfiguration" />`

File: src/Testcontainers/Configurations/Modules/Databases/MariaDbTestcontainerConfiguration.cs [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    /// </summary>
+    public MariaDbTestcontainerConfiguration()
+      : this(MariaDbImage)
+    {
+    }
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="MySqlTestcontainerConfiguration" /> class.
```
Comment (HofmeisterAn) [inline src/Testcontainers/Configurations/Modules/Databases/MariaDbTestcontainerConfiguration.cs]: `<see cref="MariaDbTestcontainerConfiguration" />`

File: src/Testcontainers/Containers/Modules/Databases/MariaDbTestcontainer.cs [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+  using DotNet.Testcontainers.Configurations;
+  using JetBrains.Annotations;
+  using Microsoft.Extensions.Logging;
+
+  /// <inheritdoc cref="TestcontainerDatabase" />
+  [PublicAPI]
+  public sealed class MariaDbTestcontainer : TestcontainerDatabase
+  {
+    /// <summary>
+    /// Initializes a new instance of the <see cref="MySqlTestcontainer" /> class.
```
Comment (HofmeisterAn) [inline src/Testcontainers/Containers/Modules/Databases/MariaDbTestcontainer.cs]: `<see cref="MariaDbTestcontainer" />`

**Review by HofmeisterAn — COMMENTED** [review]
LGTM, but I know Sonar will complain. Can't we inherit from the MySql classes (`MySqlTestcontainerConfiguration`, `MariaDbTestcontainer`)?

I can take care of the XML comments too.

**Comment by renemadsen** [comment]
> LGTM, but I know Sonar will complain.

Why will Sonar complain?

Fixed the summary comments.
Also switched to using Pomelo.EntityFrameworkCore.MySql, since it's being maintained openly on github whereas the MySql.Data is maintaned by Oracle. This affects the MySQL testing container also.

**Comment by HofmeisterAn** [comment]
> Why will Sonar complain?

Code duplication.

**Review by HofmeisterAn — APPROVED** [review]
👏 I'll merge it tomorrow — thanks. If you like, you can squash the commits and use a semantic commit message to follow the repository schema.

_I just notice that we skip those paths regarding code duplication._

**Comment by HofmeisterAn** [comment]
https://github.com/testcontainers/testcontainers-dotnet/commit/f5d471cd93797abb9a1a8108a9eae17db6d063da.

**Pull request closed without merging**